custom_components/vesync/humidifier.py

```diff
@@ -129,16 +129,22 @@ def supported_features(self):
     @property
     def target_humidity(self) -> int:
         """Return the humidity we try to reach."""
+        if type(self.smarthumidifier) is VeSyncSuperior6000S:
+            return self.smarthumidifier.details["target_humidity"]
         return self.smarthumidifier.config["auto_target_humidity"]
 
     @property
     def mode(self) -> str | None:
         """Get the current preset mode."""
+        if type(self.smarthumidifier) is VeSyncSuperior6000S:
+            return _get_ha_mode(self.smarthumidifier.mode)
         return _get_ha_mode(self.smarthumidifier.details["mode"])
 
     @property
     def is_on(self) -> bool:
         """Return True if humidifier is on."""
+        if type(self.smarthumidifier) is VeSyncSuperior6000S:
+            return True if self.smarthumidifier.device_status == "on" else False
```

Comment:
is there a reason the conditional is not just returned?

Reply:
Nope dunno why I wrote it like that 😅
I'll fix it

Reply:
Latest commit has this fix